### ISSUE TITLE
Azure disks as persistent volumes

### ISSUE THREAD

**Issue by fdipuma** [microsoft/azure-container-apps#18]
We are looking for a drop-in replacement for different services that currently can run only on VM or AKS, and the main issue seems to be the support for persistent volumes (mounts) with Azure Disks.

Long story short, we have a few services that can run inside a container but need reliable and performant storage to work properly, e.g. [Seq](https://docs.datalust.co/docs/azure-installation), [uptime-kuma](https://github.com/louislam/uptime-kuma).

Since AKS seems to be able to mount an Azure Disk as persistent storage, do you have plans on bring this feature on Container Apps?

*For context: those scenario are not currently covered by App Services or ACI because they only support mounting Azure Files / Azure Blobs volumes*

**Comment by dariagrigoriu** [comment]
@fdipuma, would Azure Files attach address the requirements as an alternative?

**Comment by fdipuma** [comment]
> @fdipuma, would Azure Files attach address the requirements as an alternative?

I'm afraid not, unfortunately Azure Files does not support workloads like databases (sqlite or similar services that heavily interact with the file system): 
https://github.com/MicrosoftDocs/azure-docs/issues/47130

It seems that in those cases only Azure Disks are supported.

**Comment by BigMorty** [comment]
Azure Files are supported. Dapr is not required

**Comment by DevSecNinja** [comment]
Would love to see support for Managed Disks as well! Some of the containers I run in AKS/Docker won't work with Azure Files.

**Comment by simonjj** [comment]
For now we have no plans to add support for this feature. If someone feels strongly please reopen/recomment.

**Comment by yuretz** [comment]
Not implementing this seriously limits the usefulness of azure container apps imo.

**Comment by Silvestre08** [comment]
Hello! 
Is there any new plans for this feature? This would greatly improve azure container apps

**Comment by mwillebrands-caesar** [comment]
@simonjj is there any way this could pop up on your radar again, as ACA is running AKS on the background, is there any way we can get Azure Container Storage supported?

**Comment by bujarsgrc** [comment]
Container Apps has huge potential — but the fact that it only supports Azure Files as persistent storage is severely limiting.

We built two designs based on Container Apps, but have now hit a major roadblock: Elasticsearch cannot reliably run on Azure Files. Because Azure Files doesn’t support the file locking, ownership, atomic operations that Lucene/Elasticsearch requires, we experience deployment failures and unusable indexes.

In short: no durable storage + filesystem incompatibility = broken indexing. We’re now seriously reconsidering using Container Apps because of this limitation.

Can Microsoft provide a better storage option (e.g. block-storage, disk mounting, enhanced volume semantics) for Container Apps — one compatible with stateful workloads like Elasticsearch?
